libtiny_tui/src/msg_area/mod.rs

```diff
@@ -159,6 +161,12 @@ impl MsgArea {
         let line_height = self.line_buf.rendered_height(self.width);
         self.lines
             .push(mem::replace(&mut self.line_buf, Line::new()));
+        // Remove first element because we went over the maximum lines
+        if let Some(max_lines) = self.max_lines {
```

Comment:
If you simplify the `Option<usize>` type this will be gone and the line below will be `if self.lines.len() > self.max_lines { .. }`.